src/scoring.jl

```diff
@@ -238,34 +239,65 @@ function ck_workers_rights(
                     ),
                 )
             end
-
-            if shift_no % WEEK_DAYS_NO != 0 && (# long break between weeks does not count
-                shifts[worker_no, shift_no] in LONG_BREAK_SEQ[1][1] &&
-                shifts[worker_no, shift_no+1] in LONG_BREAK_SEQ[1][2]
-            ) || (
-                shifts[worker_no, shift_no] in LONG_BREAK_SEQ[2][1] &&
-                shifts[worker_no, shift_no+1] in LONG_BREAK_SEQ[2][2]
-            )
-
-                long_breaks[Int(ceil(shift_no / WEEK_DAYS_NO))] = true
-            end
         end
+    end
+    return ScoringResult((penalty, errors)) + ck_workers_long_breaks(schedule_shitfs, schedule)
+end
 
-        if false in long_breaks
-            for (week_no, value) in enumerate(long_breaks)
-                if value == false
-                    penalty += penalties[string(Constraints.PEN_NO_LONG_BREAK)]
-                    @debug "The worker '$(workers[worker_no])' does not have a long break in week: '$(week_no)'"
-                    push!(
-                        errors,
-                        Dict(
-                            "code" => string(ErrorCode.LACKING_LONG_BREAK),
-                            "week" => week_no,
-                            "worker" => workers[worker_no],
-                        ),
-                    )
+function ck_workers_long_breaks(
+    schedule_shitfs::ScheduleShifts,
+    schedule::Schedule
+)::ScoringResult
+    penalty = 0
+    errors = Vector{Dict{String,Any}}()
+    workers, shifts = schedule_shitfs
+    penalties = get_penalties(schedule)
+    weeks_no = ceil(Int, size(shifts, 2) / WEEK_DAYS_NO)
+    shift_types = get_shift_options(schedule)
+
+    for week in 1:weeks_no
+        first_week_day = (week-1) * WEEK_DAYS_NO + 1
+        last_week_day = week * WEEK_DAYS_NO
+        for worker_no in axes(shifts, 1)
+            has_break = false
+            break_time = 24 - get_latest_shift_end(schedule)
```

Comment:
O co chodzi z  tym `get_latest_shift_end`, szukamy shifta brzegowego, czyli takiego, który kończy się najpóźniej? Rozumiem, że to , `break_time` może wyjść ujemny jest ok.

Reply:
Tak, get_latest_shift zwraca najpóźniejszy możliwy koniec, jeśli zmiana przechodzi przez północ, to zwracana godzina jest zwiększa o 24h. Wtedy na początku break_time może być ujemny, np. przy aktualnie domyślnych zmiana wynosi na początku -7h.

Używane są w sprawdzaniu brzegowych ciągów. Jeśli ciąg kończy się z tygodniem, to nie musi 35h, tylko przy aktualnych zmianach 28, bo w przyszłym tygodniu zmiana nie zacznie się przed 7 rano na pewno. 

Tak samo z ciągiem który zaczyna się z tygodniem, aktualnie w najgorszym wypadku zmiana skończy się 7 rano następnego dnia, więc ciąg musi to brać pod uwagę w liczbie godzin.